src/oop/special_class_method/medium/q11/MainTest.java

```diff
@@ -0,0 +1,187 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String className = "Oven";
+    private final String attributeName0 = "currTemp";
+    private final String getAttributeMethodName0 = "getCurrTemp";
+    private final String setAttributeMethodName0 = "setCurrTemp";
+    private final String attributeName1 = "size";
+    private final String getAttributeMethodName1 = "getSize";
+    private final String attributeName2 = "colour";
+    private final String getAttributeMethodName2 = "getColour";
+    private final String methodName1 = "powerDecrease";
+    private final String methodName2 = "powerIncrease";
+    private final String methodName3 = "changePower";
+    private final int changeValue = 10;
+    private final int changeLimit = 100;
+
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.medium.q11." + className;
+        this.testClass = new ObjectTest(classString);
+        String modifiedClassMessage =
+                "You have modified the provided portions of class " + className + " class in the %s attribute part. Please revert them to the original state.";
+        assertTrue(testClass.hasField(attributeName0, double.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName0));
+        assertTrue(testClass.hasField(attributeName1, double.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName1));
+        assertTrue(testClass.hasField(attributeName2, String.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName2));
+        Object[][] tests = new Object[][]{
+                {3.2, 3.4, "Red", 43243.3},
+                {432.3, 10.2, "Black", 432.2},
+                {1023.4, 11.0, "Blue", 565.4},
+                {434.3, 43.4, "White", 543.3},
+                {43.34, 7.2, "Yellow", 5555.4}
+        };
+        for (int i = 0; i < tests.length; i++) {
+            double temp = (double) tests[i][0];
+            double size = (double) tests[i][1];
+            String colour = (String) tests[i][2];
+            double updatedValue = (double) tests[i][3];
+            Object[][] arguments = {
+                    {temp, double.class},
+                    {size, double.class},
+                    {colour, String.class}
+            };
+            Class<?>[] classes = {double.class, double.class, String.class};
+            Object testInstance = testClass.createInstance(arguments);
+            _assertEquals(temp, testClass.getFieldValue(testInstance, attributeName0), modifiedClassMessage);
```

Comment:
here and every assert going down uses modifiedClassMessage which mentions an attribute as the incorrect element